src/app/store/slices/plan/index.ts

```diff
@@ -1,152 +1,141 @@
   'plan/fetchUsage',
-  async (payload: void, { getState }) => {
+  async (payload: string | undefined, { getState }) => {
     const isAuthenticated = getState().user.isAuthenticated;
 
     if (isAuthenticated) {
-      return await usageService.fetchUsage();
+      return usageService.fetchUsage(payload);
     } else {
       return null;
     }
   },
 );
 
-export const fetchSubscriptionThunk = createAsyncThunk<UserSubscription | null, void, { state: RootState }>(
-  'plan/fetchSubscription',
-  async (payload: void, { getState }) => {
-    const isAuthenticated = getState().user.isAuthenticated;
-
-    if (isAuthenticated) {
-      return paymentService.getUserSubscription();
-    } else {
-      return null;
-    }
-  },
-);
+export const fetchSubscriptionThunk = createAsyncThunk<
+  UserSubscription | null,
+  { userType: UserType },
+  { state: RootState }
+>('plan/fetchSubscription', async (payload, { getState }) => {
+  const isAuthenticated = getState().user.isAuthenticated;
+
+  if (isAuthenticated) {
+    return paymentService.getUserSubscription(payload.userType);
+  } else {
+    return null;
+  }
+});
 
 export const planSlice = createSlice({
   name: 'plan',
   initialState,
   reducers: {
-    setSubscription: (state: PlanState, action: PayloadAction<UserSubscription>) => {
-      state.subscription = action.payload;
+    setSubscriptionIndividual: (state: PlanState, action: PayloadAction<UserSubscription>) => {
+      state.individualSubscription = action.payload;
+    },
+    setSubscriptionBusiness: (state: PlanState, action: PayloadAction<UserSubscription>) => {
+      state.businessSubscription = action.payload;
     },
   },
   extraReducers: (builder) => {
     builder
-      .addCase(initializeThunk.pending, () => undefined)
+      .addCase(initializeThunk.pending, (state) => {
+        state.businessPlan = null;
+        state.businessPlanLimit = 0;
+        state.businessPlanUsage = 0;
+        state.businessPlanUsageDetails = null;
+      })
       .addCase(initializeThunk.fulfilled, () => undefined)
       .addCase(initializeThunk.rejected, () => undefined);
 
-    builder
-      .addCase(fetchPlans.pending, (state) => {
-        state.isLoadingPlans = true;
-      })
-      .addCase(fetchPlans.fulfilled, (state, action) => {
-        state.isLoadingPlans = false;
-        state.individualPlan = action.payload.individualPlan;
-        state.teamPlan = action.payload.teamPlan;
-      })
-      .addCase(fetchPlans.rejected, (state) => {
-        state.isLoadingPlans = false;
-      });
-
     builder
       .addCase(fetchLimitThunk.pending, (state) => {
         state.isLoadingPlanLimit = true;
       })
       .addCase(fetchLimitThunk.fulfilled, (state, action) => {
         state.isLoadingPlanLimit = false;
-        state.planLimit = action.payload;
+        if (!action.meta.arg) state.planLimit = action.payload;
```

Comment:
extracts the conditional variable to better underline the condition.
for example: `if(isIndividualPlan)`